tests/scorers/test_llm_integrations.py

```diff
@@ -7,76 +7,53 @@
     SummarizationScorer,
 )
 
-# Define providers and their models
-TEST_MODELS = {
-    "openai": ["gpt-4o-mini", "gpt-4o"],
-    "anthropic": ["claude-3-haiku-20240307", "claude-3-5-sonnet-20240620"],
-    "mistral": ["mistral-small-latest", "mistral-large-latest"],
-    "gemini": ["gemini-1.5-flash", "gemini-1.5-pro-latest"],
+# Centralized dictionary for providers with their associated models and API key environment variable.
+PROVIDERS = {
+    "openai": {
+        "models": ["openai/gpt-4o"],
+        "env_key": "OPENAI_API_KEY",
+    },
+    "anthropic": {
+        "models": ["anthropic/claude-3-5-sonnet-20240620"],
+        "env_key": "ANTHROPIC_API_KEY",
+    },
+    "mistral": {
+        "models": ["mistral/mistral-large-latest"],
+        "env_key": "MISTRAL_API_KEY",
+    },
+    "gemini": {
+        "models": ["gemini/gemini-1.5-pro"],
+        "env_key": "GOOGLE_API_KEY",
+    },
 }
 
-
-def get_client_and_model(provider, model):
-    api_key_env_vars = {
-        "openai": "OPENAI_API_KEY",
-        "anthropic": "ANTHROPIC_API_KEY",
-        "mistral": "MISTRAL_API_KEY",
-        "gemini": "GOOGLE_API_KEY",
-    }
-
-    if provider not in TEST_MODELS:
-        raise ValueError(f"Unknown provider: {provider}")
-
-    if model not in TEST_MODELS[provider]:
-        raise ValueError(f"Model '{model}' not available for provider '{provider}'")
-
-    api_key = os.getenv(api_key_env_vars[provider])
-    if not api_key:
-        raise OSError(
-            f"API key for {provider} not found. Please set '{api_key_env_vars[provider]}' environment variable."
-        )
-
-    if provider == "openai":
-        from openai import OpenAI
-
-        client = OpenAI(api_key=api_key)
-    elif provider == "anthropic":
-        from anthropic import Anthropic
-
-        client = Anthropic(api_key=api_key)
-    elif provider == "mistral":
-        from mistralai import Mistral
-
-        client = Mistral(api_key=api_key)
-    elif provider == "gemini":
-        import google.generativeai as genai
-
-        genai.configure(api_key=api_key)
-        client = genai.GenerativeModel(model_name=model)
-        model = "gemini"  # Adjust if necessary
-
-    return client, model
-
-
-# Generate test parameters
+# Generate test parameters by iterating over each provider and its associated models.
 test_params = [
-    (provider, model) for provider, models in TEST_MODELS.items() for model in models
+    (provider, model) for provider, cfg in PROVIDERS.items() for model in cfg["models"]
```

Comment:
This would be nicer if you used a request fixture

https://docs.pytest.org/en/7.1.x/example/simple.html#request-example

Reply:
done